ftplugin/markdown/instant-markdown.vim

```diff
@@ -158,7 +158,17 @@ function! s:bufGetLines(bufnr)
   if g:instant_markdown_autoscroll
     " inject row marker
     let row_num = max([0, line(".") - 5])
-    let lines[row_num] = join([lines[row_num], '<a name="#marker" id="marker"></a>'], ' ')
+    if len(lines) < 3
+      return lines
+    endif
+
+    "From lines[row_num] find the previous empty line that is followed by a non-empty line, or the first line.
+    while ( row_num > 0 ) && !(lines[row_num] =~ '^\s*$' && !(lines[row_num + 1] =~ '^\s*$'))
+      let row_num += -1
+    endwhile
```

Comment:
TBH, I think this while loop might be overkill. If this is the solution I wonder if we can replace this with a Vim search.